Tests/UnitTests/Core/Detector/PortalTests.cpp

```diff
@@ -168,4 +170,90 @@ BOOST_AUTO_TEST_CASE(PortalTest) {
   BOOST_CHECK_THROW(portalAI->fuse(portalBI), std::runtime_error);
 }
 
+BOOST_AUTO_TEST_CASE(PortalMaterialTest) {
+  // Volume A and B
+  auto dTransform = Acts::Transform3::Identity();
+  auto pGenerator = defaultPortalGenerator();
+  auto volumeA = DetectorVolumeFactory::construct(
+      pGenerator, tContext, "dummyA", dTransform,
+      std::make_unique<Acts::CuboidVolumeBounds>(1, 1, 1),
+      tryAllPortalsAndSurfaces());
+  auto volumeB = DetectorVolumeFactory::construct(
+      pGenerator, tContext, "dummyB", dTransform,
+      std::make_unique<Acts::CuboidVolumeBounds>(1, 1, 1),
+      tryAllPortalsAndSurfaces());
+
+  // Create some material
+  auto materialSlab = Acts::MaterialSlab(
+      Acts::Material::fromMolarDensity(1., 2., 3., 4., 5.), 1.);
+  auto materialA =
+      std::make_shared<Acts::HomogeneousSurfaceMaterial>(materialSlab);
+  auto materialB =
+      std::make_shared<Acts::HomogeneousSurfaceMaterial>(materialSlab);
+
+  // A few portals
+  auto rectangle = std::make_shared<Acts::RectangleBounds>(10., 100.);
+
+  auto surfaceA = Acts::Surface::makeShared<Acts::PlaneSurface>(
+      Acts::Transform3::Identity(), rectangle);
+  surfaceA->assignSurfaceMaterial(materialA);
+  auto portalA = Acts::Experimental::Portal::makeShared(surfaceA);
+
+  DetectorVolumeUpdator linkToA;
+  auto linkToAImpl = std::make_unique<const LinkToVolumeImpl>(volumeA);
+  linkToA.connect<&LinkToVolumeImpl::link>(std::move(linkToAImpl));
+  portalA->assignDetectorVolumeUpdator(Acts::Direction::Positive,
+                                       std::move(linkToA), {volumeA});
+
+  auto surfaceB = Acts::Surface::makeShared<Acts::PlaneSurface>(
+      Acts::Transform3::Identity(), rectangle);
+  auto portalB = Acts::Experimental::Portal::makeShared(surfaceB);
+  DetectorVolumeUpdator linkToB;
+  auto linkToBImpl = std::make_unique<const LinkToVolumeImpl>(volumeB);
+  linkToB.connect<&LinkToVolumeImpl::link>(std::move(linkToBImpl));
+  portalB->assignDetectorVolumeUpdator(Acts::Direction::Negative,
+                                       std::move(linkToB), {volumeB});
+
+  // Portal A fuses with B
+  // - has material and keeps it, portal B becomes portal A
+  portalA->fuse(portalB);
+  BOOST_CHECK_EQUAL(portalA->surface().surfaceMaterial(), materialA.get());
+  BOOST_CHECK_EQUAL(portalA, portalB);
```

Comment:
```suggestion
  // Portal A fuses with B
  // - has material and keeps it, portal B becomes portal A
  BOOST_REQUIRE_NE(portalA, portalB)
  portalA->fuse(portalB);
  BOOST_CHECK_EQUAL(portalA->surface().surfaceMaterial(), materialA.get());
  BOOST_CHECK_EQUAL(portalA, portalB);
```
Could it make sense to check before the fusing that the portals differ and the fusing actually changes their state?